cmd/context/context.go

```diff
@@ -222,7 +217,40 @@ func Init(ctx context.Context) error {
 			return err
 		}
 	}
+	if okteto.IsOktetoContext() {
+		client, err := okteto.NewOktetoClient()
+		if err != nil {
+			return err
+		}
+		secretsAndKubeCredentials, err := client.GetSecretsAndKubeCredentials(ctx)
+		if err != nil {
+			return err
+		}
+
+		if err := updateContext(secretsAndKubeCredentials.Credentials); err != nil {
+			return err
+		}
+		setSecrets(secretsAndKubeCredentials.Secrets)
+	}
+
 	os.Setenv("OKTETO_USERNAME", okteto.Context().Username)
```

Comment:
I wouldn't do that because we call `okteto.IsOktetoContext()` more than once for some commands, I think injecting them at the beginning of the command and not calling the api for retrieving the secrets again should be the path to go

Reply:
I mean to the `if` section here:
https://github.com/okteto/okteto/pull/1800/files#diff-7ba016dd78fdd00cce76cf854dcc18447a4228dc8dc8bcd660fcdd68876d38e9R229